map/bookmark_manager.hpp

```diff
@@ -277,6 +277,7 @@ class BookmarkManager final
   kml::MarkGroupId GetCategoryId(std::string const & name) const;
 
   kml::GroupIdCollection const & GetBmGroupsIdList() const { return m_bmGroupsIdList; }
+  kml::GroupIdCollection GetSortedBmGroupIdList() const;
```

Comment:
How often is this method called? Does it make sense to sort and cache sorted lists in m_bmGroupsIdList, to avoid resorting it again and again each time for displaying?

Does it make sense to insert sorting into each bookmark/track modification operation (e.g. sort when any category has updated its last modified date)? Then there is no need to sort list of categories each time when retrieving it. By the way, the sorting by the last modified field in such cases for the vector can be implemented without a full sorting: scan vector to find the index of the modified category, and then move it to the beginning of the vector.